src/rust/engine/process_execution/src/lib.rs

```diff
@@ -341,13 +344,70 @@ pub struct ProcessMetadata {
 ///
 /// The result of running a process.
 ///
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Derivative, Clone, Debug, Eq)]
+#[derivative(PartialEq, Hash)]
 pub struct FallibleProcessResultWithPlatform {
   pub stdout_digest: Digest,
   pub stderr_digest: Digest,
   pub exit_code: i32,
-  pub platform: Platform,
   pub output_directory: hashing::Digest,
+  pub platform: Platform,
+  #[derivative(PartialEq = "ignore", Hash = "ignore")]
+  pub metadata: ProcessResultMetadata,
+}
+
+/// Metadata for a ProcessResult corresponding to the REAPI `ExecutedActionMetadata` proto. This
+/// conversion is lossy, but the interesting parts are preserved.
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct ProcessResultMetadata {
+  pub execution_time: Option<Duration>,
+}
+
+impl ProcessResultMetadata {
+  pub fn new(execution_time: Option<Duration>) -> Self {
+    ProcessResultMetadata { execution_time }
+  }
+}
+
+impl From<ExecutedActionMetadata> for ProcessResultMetadata {
+  fn from(metadata: ExecutedActionMetadata) -> Self {
+    let execution_time = match (
+      metadata.execution_start_timestamp,
+      metadata.execution_completed_timestamp,
+    ) {
+      (Some(started), Some(completed)) => TimeSpan::from_start_and_end(&started, &completed, "")
```

Comment:
The string is used for `Err`, but we then throw that away with `.ok()` so I left it empty.

Reply:
That's a weird API... ideally it would return a non-allocated error type with a `Debug`/`Display` impl, rather than passing in a string to format. But yea, this is fine for now.

We might consider logging at debug for negative times though...